index.js

```diff
@@ -2,73 +2,119 @@
 const errors = require('restify-errors')
 const joi = require('joi')
 
-const defaultErrorTransformer = (validationInput, joiError) => {
-  const retError = new errors.BadRequestError()
-  retError.body.message = joiError.message
-  retError.body.data = joiError.details
-  return retError
+const middlewareDefaults = {
+  joiOptions: {
+    convert: true,
+    allowUnknown: false,
+    abortEarly: false
+  },
+  errorTransformer: (validationInput, joiError) => {
+    const retError = new errors.BadRequestError()
+    retError.body.message = joiError.message
+    retError.body.data = joiError.details
+    return retError
+  },
+  errorResponder: (transformedErr, req, res, next) => next(transformedErr),
+  keysToValidate: ['params', 'body', 'query', 'user', 'headers', 'trailers', 'files']
+}
+
+const optionsSchema = {
+  joiOptions: joi.object(),
+  errorTransformer: joi.func(),
+  errorResponder: joi.func(),
+  keysToValidate: joi.array().items(joi.string())
+}
+
+const checkOptions = (defaults, input) => {
```

Comment:
This is used to check both the middleware options as well as the per route overrides